rixs/process2d.py

```diff
@@ -0,0 +1,240 @@
+
+    Parameters
+    -------------
+    photon_events : array
+        two column x, y photon location coordinates
+    curvature : array
+        The polynominal coeffcients describing the image curvature.
+        These are in decreasing order e.g.
+        curvature[0]*x^2 + curvature[1]*x^1, + curvature[2]*x^0, `
+    bins : int or array_like or [int, int] or [array, array]
+        The bin specification in y then x order:
+            * If int, the number of bins for the two dimensions (nx=ny=bins).
+            * If array_like, the bin edges for the two dimensions
+              (y_edges=x_edges=bins).
+            * If [int, int], the number of bins in each dimension
+              (ny, nx = bins).
+            * If [array, array], the bin edges in each dimension
+              (y_edges, x_edges = bins).
+            * A combination [int, array] or [array, int], where int
+              is the number of bins and array is the bin edges.
+
+    Returns
+    -----------
+
+    """
+    x_centers, offsets = get_curvature_offsets(photon_events, bins)
+    reversed_curvature = polyfit(x_centers, offsets, len(guess_curvature) - 1)
+    curvature = reversed_curvature[::-1]
+
+    return curvature
+
+
+def estimate_elastic_pos(photon_events, x_range=(0, 20), bins=None):
+    """
+    Estimate where the elastic line is.
+
+
+    Parameters
+    ----------
+    photon_events : array
+        two column x, y, Iph photon locations and intensities
+    x_range : (float, float)
+        Select x region of image to use
+        A tuple defines a range in x (minx, maxx)
+    bins : int or sequence of scalars or str, optional
+        Binning in the y direction.
+        If 'bins' is None a step of 1 is assumed over the relavant range
+        If `bins` is an int, it defines the number of equal-width
+        bins in the given range (10, by default). If `bins` is a
+        sequence, it defines the bin edges, including the rightmost
+        edge, allowing for non-uniform bin widths.
+    """
+    x = photon_events[:, 0]
+    y = photon_events[:, 1]
+    Iph = photon_events[:, 2]
+
+    choose = np.logical_and(x > x_range[0], x < x_range[1])
+    y_choose = y[choose]
+    I_choose = Iph[choose]
+
+    if bins is None:
+        bins = np.arange(y_choose.min()//1 + 0.5, y_choose.max()//1 - 0.5)
+    I_profile, y_edges = np.histogram(y_choose, bins=bins, weights=I_choose)
+    y_centers = (y_edges[:-1] + y_edges[1:])/2
+    elastic_y_value = y_centers[np.argmax(I_profile)]
+    return elastic_y_value
+
+
+def extract(photon_events, curvature, bins=None):
+    """Apply curvature to photon events to create pixel versus intensity spectrum
+
+    Parameters
+    ----------
+    photon_events : array
+        three column x, y , Iph photon locations and intensities
+    curvature : array
+        The polynominal coeffcients describing the image curvature.
+        These are in decreasing order e.g.
+    curvature[0]*x^2 + curvature[1]*x^1, + curvature[2]*x^0,
```

Comment:
I think placing this inside a code-block will be a good idea: this can be done using the syntax:
.. code-block:: python
    curvature[0]*x^2 + curvature[1]*x^1 + curvature[2]*x^0

Alternatively it could be done inside as math, with the same syntax as above but with 'code-block:: python' replaced with 'math::'

Reply:
Will do. Since this line is now code rather than just an illustration. I will change ^ to **   

    .. code-block:: python
       curvature[0]*x**2 + curvature[1]*x**1 + curvature[2]*x**0